### PR TITLE
Add support for visiting fragment definitions in QueryVisitor.

### PR DESCRIPTION
**Pull request by tsroka**

### CODE REVIEW

File: src/main/java/graphql/analysis/QueryVisitor.java [hunk 1]
```diff
@@ -16,4 +16,8 @@
 
     void visitFragmentSpread(QueryVisitorFragmentSpreadEnvironment queryVisitorFragmentSpreadEnvironment);
 
+    default void visitFragment(QueryVisitorFragmentEnvironment queryVisitorFragmentEnvironment) {
```
Comment (andimarek) [inline src/main/java/graphql/analysis/QueryVisitor.java]: I think we should call it `visitFragmentDefinition`. `visitFragment` is not clear enough.

File: src/main/java/graphql/analysis/QueryVisitorFragmentEnvironment.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package graphql.analysis;
+
+import graphql.PublicApi;
+import graphql.language.FragmentDefinition;
+import graphql.language.Node;
+import graphql.util.TraverserContext;
+
+@PublicApi
+public interface QueryVisitorFragmentEnvironment {
```
Comment (andimarek) [inline src/main/java/graphql/analysis/QueryVisitorFragmentEnvironment.java]: same as above: `QueryVisitorFragmentDefitintionEnvironment` is more accurate and more future proof.

File: src/main/java/graphql/analysis/QueryVisitorFragmentEnvironmentImpl.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+import graphql.Internal;
+import graphql.language.FragmentDefinition;
+import graphql.language.Node;
+import graphql.util.TraverserContext;
+
+import java.util.Objects;
+
+@Internal
+public class QueryVisitorFragmentEnvironmentImpl implements QueryVisitorFragmentEnvironment {
```
Comment (andimarek) [inline src/main/java/graphql/analysis/QueryVisitorFragmentEnvironmentImpl.java]: a toString would be nice, but is not strictly needed.

**Comment by tsroka** [comment]
> nice ... see my comments about naming

Renamed as per your comments and added #toString

File: src/test/groovy/graphql/analysis/QueryTraversalTest.groovy [hunk 4]
```diff
@@ -20,7 +21,7 @@ import static graphql.schema.GraphQLList.list
 import static graphql.schema.GraphQLNonNull.nonNull
 import static java.util.Collections.emptyMap
 
-class QueryTraversalTest extends Specification {
+class QueryTrqaversalTest extends Specification {
```
Comment (andimarek) [inline src/test/groovy/graphql/analysis/QueryTraversalTest.groovy]: typo :-)

**Pull request merged**